x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/models_management/models_list.tsx

```diff
@@ -0,0 +1,469 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useCallback, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  Direction,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiTitle,
+  EuiButton,
+  EuiSearchBarProps,
+  EuiSpacer,
+  EuiButtonIcon,
+  EuiBadge,
+} from '@elastic/eui';
+// @ts-ignore
+import { formatDate } from '@elastic/eui/lib/services/format';
+import { EuiBasicTableColumn } from '@elastic/eui/src/components/basic_table/basic_table';
+import { EuiTableSelectionType } from '@elastic/eui/src/components/basic_table/table_types';
+import { Action } from '@elastic/eui/src/components/basic_table/action_types';
+import { StatsBar, ModelsBarStats } from '../../../../../components/stats_bar';
+import { useInferenceApiService } from '../../../../../services/ml_api_service/inference';
+import { ModelsTableToConfigMapping } from './index';
+import { TIME_FORMAT } from '../../../../../../../common/constants/time_format';
+import { DeleteModelsModal } from './delete_models_modal';
+import { useMlKibana, useNotifications } from '../../../../../contexts/kibana';
+import { ExpandedRow } from './expanded_row';
+import { getResultsUrl } from '../analytics_list/common';
+import {
+  ModelConfigResponse,
+  ModelPipelines,
+  TrainedModelStat,
+} from '../../../../../../../common/types/inference';
+import {
+  REFRESH_ANALYTICS_LIST_STATE,
+  refreshAnalyticsList$,
+  useRefreshAnalyticsList,
+} from '../../../../common';
+
+type Stats = Omit<TrainedModelStat, 'model_id'>;
+
+export type ModelItem = ModelConfigResponse & {
+  type?: string;
+  stats?: Stats;
+  pipelines?: ModelPipelines['pipelines'] | null;
+};
+
+export type ModelItemFull = Required<ModelItem>;
+
+export const ModelsList: FC = () => {
+  const {
+    services: {
+      application: { navigateToUrl },
+    },
+  } = useMlKibana();
+  const inferenceApiService = useInferenceApiService();
+  const { toasts } = useNotifications();
+
+  const [searchQueryText, setSearchQueryText] = useState('');
+
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [sortField, setSortField] = useState<string>(ModelsTableToConfigMapping.id);
+  const [sortDirection, setSortDirection] = useState<Direction>('asc');
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [items, setItems] = useState<ModelItem[]>([]);
+  const [selectedModels, setSelectedModels] = useState<ModelItem[]>([]);
+
+  const [modelsToDelete, setModelsToDelete] = useState<ModelItemFull[]>([]);
+
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<Record<string, any>>({});
+
+  // Subscribe to the refresh observable to trigger reloading the model list.
+  useRefreshAnalyticsList({
+    isLoading: setIsLoading,
+    onRefresh: () => fetchData(),
```

Comment:
Could you explain why `fetchData` needs to be wrapped in an arrow function?

Reply:
I just copy-pasted this block of code from the analytics jobs component and changed the function call. There is definitely no need to wrap it, will change  🙂

Reply:
Changed in https://github.com/elastic/kibana/pull/74978/commits/2fde1adf32a219358ab87a7c0bcf24f4918313e3